addons/web/models/ir_http.py

```diff
@@ -62,9 +62,16 @@ def session_info(self):
                 "qweb": qweb_checksum,
                 "translations": translation_hash,
             }
+            company_ids = user.company_id
+            if request and hasattr(request, 'httprequest'):
+                company_ids = request.httprequest.cookies.get('cids', [str(user.company_id.id)]).split(",")
             session_info.update({
                 # current_company should be default_company
-                "user_companies": {'current_company': (user.company_id.id, user.company_id.name), 'allowed_companies': [(comp.id, comp.name) for comp in user.company_ids]},
+                "user_companies": {
+                    'current_company': (user.company_id.id, user.company_id.name),
+                    'allowed_companies': [(comp.id, comp.name) for comp in user.company_ids],
+                    "company_ids": company_ids,
```

Comment:
In general you shouldn't be modifying `ir_http.py`, `http.py` or any other low-level server-side component unless you have a very good reason to ;-)

Here, it's not clear to me why that change is necessary, and worse, it seems to be introducing confusing notions of what companies are currently enabled on the server-side: the `allowed_companies` value may be different from your new `company_ids` value.

If I remember correctly our design [1], it's the client-side who is responsible for parsing the `cids` and determining the appropriate `allowed_company_ids` value to pass in the context. And there are already several places where it's doing that, so there should be no need to copy that logic server-side.


[1] See a5b6f31cf28e5381e1c85f66730bcdb55998e643